### PR TITLE
Add tests rdbstorage in github actions

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Motivation

To replace [tests-rdbstorage in CircleCI](https://github.com/optuna/optuna/blob/master/.circleci/config.yml#L51-L110) by this one.

## Description of the changes

Run the same tests by github actions.

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@hvy Could you review this PR if you have time?

File: .github/workflows/tests-rdbstorage.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+name: Tests (RDB storage)
+
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+  schedule:
+    - cron: '0 23 * * SUN-THU'
```
Comment (keisuke-umezawa) [inline .github/workflows/tests-rdbstorage.yml:9]: is this schedule is expected?

File: .github/workflows/tests-rdbstorage.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+jobs:
+  tests-rdbstorage:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
```
Comment (keisuke-umezawa) [inline .github/workflows/tests-rdbstorage.yml:17]: Do we need to run this test for every python version?

Comment (keisuke-umezawa) [reply .github/workflows/tests-rdbstorage.yml:17]: I will run it also 3.9.

File: .github/workflows/tests-rdbstorage.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        env:
+          MYSQL_ROOT_PASSWORD: mandatory_arguments
+          MYSQL_DATABASE: optunatest
+          MYSQL_USER: user
+          MYSQL_PASSWORD: test
+        options: >-
+          --health-cmd "mysqladmin ping -h localhost"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
```
Comment (keisuke-umezawa) [inline .github/workflows/tests-rdbstorage.yml:33]: I will add why I added this health check.

File: .github/workflows/tests-rdbstorage.yml (hunk 1 above)
Comment (HideakiImamura) [reply .github/workflows/tests-rdbstorage.yml:9]: I think this schedule is appropriate. The test is executed every weekday. Do you have any concerns?

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR! The added test looks solid to me. I'm looking forward to your update as you noted.

**Comment by crcrpar** [comment]
Do you want to remove rdbstorage test from [.circleci/config.yml](https://github.com/optuna/optuna/blob/master/.circleci/config.yml#L8) and update https://github.com/optuna/optuna/blob/master/CONTRIBUTING.md#continuous-integration-and-local-verification?

File: .github/workflows/tests-rdbstorage.yml (hunk 2 above)
Comment (hvy) [reply .github/workflows/tests-rdbstorage.yml:17]: Sounds good. I'd expect that to be a requirement.

File: .github/workflows/tests-rdbstorage.yml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    services:
+      mysql:
+        image: mysql:5.7
+        ports:
+          - 3306:3306
+        env:
+          MYSQL_ROOT_PASSWORD: mandatory_arguments
```
Comment (hvy) [inline .github/workflows/tests-rdbstorage.yml:25]: Just curious but is this variable introduced for the same reason as what's discussed in https://github.community/t/mysql-service-never-comes-up-healthy-in-action/16890/10 ?

**Review by hvy — COMMENTED** [review]
Looks solid and LGTM besides those already ongoing discussions about Python 3.9 and the comment by @crcrpar .

File: .github/workflows/tests-rdbstorage.yml (hunk 1 above)
Comment (keisuke-umezawa) [reply .github/workflows/tests-rdbstorage.yml:9]: No concerns! Thanks.

File: .github/workflows/tests-rdbstorage.yml (hunk 3 above)
Comment (keisuke-umezawa) [reply .github/workflows/tests-rdbstorage.yml:33]: Most examples use health checks. https://freek.dev/1590-how-to-use-a-mysql-database-on-github-actions
The benefit is that we can find an error at the "Initialize containers" step if containers are not running correctly.

**Comment by keisuke-umezawa** [comment]
@crcrpar 

> Do you want to remove rdbstorage test from .circleci/config.yml and update https://github.com/optuna/optuna/blob/master/CONTRIBUTING.md#continuous-integration-and-local-verification?

After merging this PR, I will make this check required and make the above follow-up PR.

File: .github/workflows/tests-rdbstorage.yml (hunk 4 above)
Comment (keisuke-umezawa) [reply .github/workflows/tests-rdbstorage.yml:25]: You can find `MYSQL_ROOT_PASSWORD` is mandatory if you check [this page](https://hub.docker.com/_/mysql).

**Comment by keisuke-umezawa** [comment]
@hvy @HideakiImamura 
Sorry, tests with 3.9 failed. 87d2aa8

Can I work on it as another follow up PR?

File: .github/workflows/tests-rdbstorage.yml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install --progress-bar off -U setuptools
+
+        # Install minimal dependencies and confirm that `import optuna` is successful.
+        pip install --progress-bar off .
+        python -c 'import optuna'
+        optuna --version
+
+        pip install --progress-bar off .[testing] -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (hvy) [inline .github/workflows/tests-rdbstorage.yml:85]: Sorry for the late comment but do we need all integration modules? The failure with Python 3.9 seemingly comes from trying to install `tensorflow` and I think it'll be fixed by this change as well.
```suggestion
        pip install --progress-bar off .[tests]
```

Comment (keisuke-umezawa) [reply .github/workflows/tests-rdbstorage.yml]: Thank you! I try it.

**Review by hvy — APPROVED** [review]
LGTM!

I'm fine either way updating the CircleCI here or in a follow up. By the way, we could make this workflow a merge requirement. I think it's quite fundamental and it shouldn't be a bottleneck in the merge process.

**Review by not522 — APPROVED** [review]
Thanks, LGTM!

**Comment by keisuke-umezawa** [comment]
Follow up PR: https://github.com/optuna/optuna/pull/2255

**Pull request merged**